### ISSUE TITLE
WebDAV forces folders and files to lowercase

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#17698]
When you upload files and folders to webdav, it forces all files and folder names to lowercase.  This is a remnant of the the misguided work in 4.x to lowercase all file and folder names.

File and folder names should properly reflect their case when being saved and shown
AND 
will be case-insensitive when being requested via URL.

To test:

Create a folder and file on your desktop called 
```
./TESTING
./TESTING/My File.txt
```
upload them both - they should be shown with the proper casing.

**Comment by jdotcms** [comment]
[PR](https://github.com/dotCMS/core/pull/17713/files)

**Comment by jdotcms** [comment]
The file is being uploaded with uppercase, but search is insensitive
<img width="561" alt="Screen Shot 2019-12-06 at 10 03 50 AM" src="https://user-images.githubusercontent.com/19578350/70337311-6b7b2f80-1810-11ea-9647-8223fb013680.png">
<img width="534" alt="Screen Shot 2019-12-06 at 10 03 37 AM" src="https://user-images.githubusercontent.com/19578350/70337312-6b7b2f80-1810-11ea-82ba-994af2b0582a.png">
<img width="600" alt="Screen Shot 2019-12-06 at 10 03 24 AM" src="https://user-images.githubusercontent.com/19578350/70337313-6b7b2f80-1810-11ea-8507-82e35c5d6346.png">

**Comment by fmontes** [comment]
There are some issues with rename that @jdotcms is investigating.

We also needs to test the whole CRUD.

**Comment by wezell** [comment]
Please only test/fix renaming issues in cyberduck.  I do not want to waste time hunting down obscure issues with Finder's webdav support.

**Comment by wezell** [comment]
Moving back to the bug sprint - they can complete the work.

**Comment by freddyucv** [comment]
Tested:
- Renaming file
- Deleting file
- Editing File

**Comment by dotcms-corp** [comment]
Please test folders too

**Comment by fabrizzio-dotCMS** [comment]
Please correct me if I'm wrong cuz these fixes do not seem to be in master.
Moreover, There's a PR referenced from this issue that hasn't been merged yet? 
Can somebody please tell me where the internal QA was done?

**Comment by wezell** [comment]
@fabrizzio-dotCMS I don't think they are in master.  They are here:
https://github.com/dotCMS/core/compare/issue-17698-donotforce-lower-webdav

**Comment by fabrizzio-dotCMS** [comment]
There's a huge bug so that if you attempt to rename a folder to upper case it. 
let's say 
/demo.dotcms.com/propducts to be /demo.dotcms.com/Propducts 
There's a flag like **deleteExistingBeforeMove** in the MoveHandler class that instructs webdav to remove the current folder so the next time a hit to locate the folder is triggered results in an Exception preventing the rename process from finishing. 

Long story short.. if you rename a folder to upper-case it will result in a folder loss. The folder will be removed without further notice. The Bug can be reproduced in demo.

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/17802
